forte/data/readers/classification_reader.py

```diff
@@ -0,0 +1,245 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import csv
+import importlib
+from typing import Iterator, Tuple, List
+from forte.common import Resources, ProcessorConfigError
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.base_reader import PackReader
+from ft.onto.base_ontology import *
+
+
+__all__ = ["ClassificationDatasetReader"]
+
+
+class ClassificationDatasetReader(PackReader):
+    r"""class:`ClassificationDatasetReader` is designed to read text
+    classification dataset that contains input text and digit/text labels.
+
+    User must specify `forte_data_fields` which is a list of forte ontologies
+    or labels and it should be one-to-one correspondence with the original data
+    fields. This configuration helps generating labels and initializing data
+    fields within corresponding ontologies wrappers.
+    For example, for amazon polarity dataset,
+    https://huggingface.co/datasets/amazon_polarity, the original data fields
+    are [label, title, content]. We want to identify them as either class
+    label or specific ontologies by specifying
+    "forte_data_fields: ['label', 'ft.onto.base_ontology.Title',
+    'ft.onto.ag_news.Description']" in the configuration.
+
+    User must also specify `input_ontologies` that will be concatenate as
+    input text in a list.
+    The number of input ontologies and the order of concatenation
+    can be customized in the list. For example, if we only want titles and
+    descriptions in our input text, we specify "input_ontologies":
+    ['ft.onto.base_ontology.Title', 'ftx.onto.ag_news.Description']" in the
+    configuration. We can also only include descriptions by specifying
+    "input_ontologies":['ftx.onto.ag_news.Description']" which is very
+    flexible in customizing input text.
+
+
+    User must also specify `index2class` which is the mapping from zero-based
```

Comment:
why zero-based, what if my dataset is not zero based

Reply:
Keys are only used internally and values are `class_names` from dataset. The dataset class names can be any digit/text that is stored in values.